src/inspectors/CollaborationInspector.ts

```diff
@@ -39,4 +46,40 @@ export class CollaborationInspector implements ICollaborationInspector {
   async getPullsDiffStat(owner: string, repo: string, prNumber: string) {
     return this.service.getPullsDiffStat(owner, repo, prNumber);
   }
+
+  private async getMaxNumberOfPullRequests(
+    owner: string,
+    repo: string,
+    options: { withDiffStat?: boolean; maxNumberOfPullRequests?: number } & ListGetterOptions<{ state?: PullRequestState }>,
+  ): Promise<Paginated<PullRequest>> {
+    let items: PullRequest[] = [];
+    let hasNextPage = true;
+    let i = 1;
+
+    while (hasNextPage) {
+      const pullRequests = await this.service.getPullRequests(owner, repo, {
+        ...options,
+        ...{ pagination: { page: i, perPage: options.pagination?.perPage } },
+      });
+      //Add pull requstes to the existing array of PRs from another page
+      items = _.merge(items, pullRequests.items);
+
+      if (<number>options?.maxNumberOfPullRequests >= items.length) {
+        //Get maximum n newest pull requests
+        items = _.take(items, options.maxNumberOfPullRequests);
+        break;
+      }
+      hasNextPage = pullRequests.hasNextPage;
+      i++;
+    }
+
+    return {
+      hasNextPage: false,
+      hasPreviousPage: false,
+      items,
+      totalCount: items.length,
+      page: 1,
+      perPage: items.length,
+    };
+  }
```

Comment:
note: I don't like this function. I am working on a generic function to be able to use it for other functions too.